jsapp/js/components/submissions/submissionUtils.ts

```diff
@@ -396,17 +449,46 @@ function getRegularGroupAnswers(data, targetKey) {
   return answers;
 }
 
+function getRowListName(row: SurveyRow | undefined): string | undefined {
+  let returnVal;
+  if (row && Object.keys(row).includes(CHOICE_LISTS.SELECT)) {
+    returnVal = row[CHOICE_LISTS.SELECT as keyof SurveyRow];
+  }
+  if (row && Object.keys(row).includes(CHOICE_LISTS.MATRIX)) {
+    returnVal = row[CHOICE_LISTS.MATRIX as keyof SurveyRow];
+  }
+  if (row && Object.keys(row).includes(CHOICE_LISTS.SCORE)) {
+    returnVal = row[CHOICE_LISTS.SCORE as keyof SurveyRow];
+  }
+  if (row && Object.keys(row).includes(CHOICE_LISTS.RANK)) {
+    returnVal = row[CHOICE_LISTS.RANK as keyof SurveyRow];
+  }
+  if (typeof returnVal === 'string') {
+    return returnVal;
+  }
+  return undefined;
+}
+
 /**
- * @param {object} row
- * @returns {string|undefiend}
+ * Returns an attachment object or an error message.
  */
-function getRowListName(row) {
-  return (
-    row[CHOICE_LISTS.SELECT] ||
-    row[CHOICE_LISTS.MATRIX] ||
-    row[CHOICE_LISTS.SCORE] ||
-    row[CHOICE_LISTS.RANK]
+export function getMediaAttachment(
+  submission: SubmissionResponse,
+  fileName: string
+): string | SubmissionAttachment {
+  const fileNameNoSpaces = fileName.replace(/ /g, '_');
+  let mediaAttachment: string | SubmissionAttachment = t('Could not find ##fileName##').replace(
```

Comment:
I wonder if it would be better practice to do something like:
```
const attachmentNotFound = t('Could not find ##fileName##').replace(...)
let mediaAttachment: SubmissionAttachment
...
if (mediaAttachment) {...} else return attachmentNotFound
```
to ensure that `mediaAttachment`'s typing? Not a big deal, more curious :D

Reply:
I tried doing that now, but it turned out I need to do some other stuff around to make it work, so I think it's ok to leave it as is.